ppl/zqd/storage/filestore/filestore.go

```diff
@@ -217,6 +233,112 @@ func (s *Storage) syncInfoFile() error {
 		}
 		return nil
 	}
-	info := info{s.span.Ts, s.span.End()}
+	info := info{
+		MinTime:       s.span.Ts,
+		MaxTime:       s.span.End(),
+		AlphaMigrated: true,
+	}
 	return fs.MarshalJSONFile(info, path, 0600)
 }
+
+func (s *Storage) migrateAlphaZngIfNeeded(ctx context.Context) error {
+	if s.alphaMigrated {
+		return nil
+	}
+	if err := s.wsem.Acquire(ctx, 1); err != nil {
+		return err
+	}
+	defer s.wsem.Release(1)
+	exists, err := s.ensureAllZngFile()
+	if err != nil {
+		return err
+	}
+	if !exists {
+		s.alphaMigrated = true
+		return nil
+	}
+	isAlpha, err := isAlphaZngFile(s.join(allZngFile))
+	if err != nil {
+		return err
+	}
+	if !isAlpha {
```

Comment:
Since migrateFile also does a syncInfoFile it seems like the syncInfoFile should go here after a successful migration.